android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/KeyboardController.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (C) 2020 SIL International. All rights reserved.
+ */
+
+package com.tavultesoft.kmea.data;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.tavultesoft.kmea.JSONParser;
+import com.tavultesoft.kmea.data.Keyboard;
+import com.tavultesoft.kmea.KeyboardPickerActivity;
+import com.tavultesoft.kmea.util.FileUtils;
+import com.tavultesoft.kmea.util.MapCompat;
+import com.tavultesoft.kmea.KMManager;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.ObjectInputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class KeyboardController {
+  public static final String TAG = "InstalledKBList";
+  public static final String KMFilename_Installed_KeyboardsList = "keyboards_list.json";
+
+  private static KeyboardController instance;
+
+  /**
+   * @return get or create shared instance.
+   */
+  public static KeyboardController getInstance() {
+    if (instance == null) {
+      createInstance();
+    }
+    return instance;
+  }
+
+  /**
+   * create singleton instance.
+   */
+  private synchronized static void createInstance()
```

Comment:
This can probably just be folded into `getInstance()` and then we avoid the extra code...